src/psycopmlutils/timeseriesflattener/flattened_dataset.py

```diff
@@ -73,18 +76,19 @@ def __init__(
         # in self.df
         self.df_aggregating = self.pred_times_with_uuid
         self.df = self.pred_times_with_uuid
```

Comment:
Depending on how these are used later on, might run into issues of shared memory, i.e. modifying `self.pred_times_with_uuid` might also modify `self.df` and vice versa. https://stackoverflow.com/questions/27673231/why-should-i-make-a-copy-of-a-data-frame-in-pandas

Reply:
Ah, that's excellent, thanks! Have added a .copy() where relevant.